pymodm/fields.py

```diff
@@ -283,7 +283,10 @@ def __init__(self, verbose_name=None, mongo_name=None,
 
         def validate_min_and_max(value):
             # Turn value into a Decimal.
-            value = value.to_decimal()
+            if isinstance(value, Decimal128):
+                value = value.to_decimal()
+            elif not isinstance(value, decimal.Decimal):
+                value = decimal.Decimal(value)
```

Comment:
Was this to fix a failing test? I thought that validators also received the `to_python`ed values?

Reply:
We see the following failure without this:
```
======================================================================
FAIL: test_validate (test.field_types.test_decimal128_field.Decimal128FieldTestCase)
----------------------------------------------------------------------
pymodm.errors.ValidationError: ["'int' object has no attribute 'to_decimal'"]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/pmital/Developer/pymodm/test/field_types/test_decimal128_field.py", line 54, in test_validate
    self.field.validate(101)
AssertionError: "greater than maximum" does not match "["'int' object has no attribute 'to_decimal'"]"

----------------------------------------------------------------------
```

Reply:
The reason this happens is because I removed a call to `to_python` from `MongoBaseField.validate` ([link](https://github.com/mongodb/pymodm/pull/41/files#diff-fbfb6c1a9962c447bda1cb9ced97fcb9L172)). The rationale for doing so is that validation - by itself - should not make calls to `to_python`, otherwise multiple attempts to validate a certain field will result in repeated type casting operations. Does that seem reasonable?

Reply:
I'm guessing that means we cannot have this change in `0.4.1` as we don't want breaking changes? I am not sure how we can avoid it at this point...